lib/takeWhile.js

```diff
@@ -7,7 +7,11 @@
  * @returns {Array} slice of the array
  */
 function takeWhile(ary, predicate) {
-  throw new Error("Method is still a skeleton");
+  if (arguments.length != 2 || !Array.isArray(ary) || typeof predicate != "function")
+    throw new Error();
+
+  const sliceIndex = ary.findIndex(element => !predicate(element));
```

Comment:
`findIndex` isn't supported in ie. Could we use the better supported `indexOf` instead?

Reply:
Yep, my bad. Don't know how I could use `indexOf` for that though, I'll probably use a plain loop instead. Thanks @ktilcu